libminifi/src/utils/OsUtils.cpp

```diff
@@ -284,6 +284,22 @@ int64_t OsUtils::getTotalPagingFileSize() {
   DWORDLONG total_paging_file_size = memory_info.ullTotalPageFile;
   return total_paging_file_size;
 }
+
+std::string OsUtils::getWindowsErrorAsString(DWORD error_code) {
+  if (error_code == 0)
+    return "";
+
+  LPSTR message_buffer = nullptr;
+
+  size_t size = FormatMessageA(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                               nullptr, error_code, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                               (LPSTR)&message_buffer, 0, nullptr);
+
+  std::string message(message_buffer, size);
+
+  LocalFree(message_buffer);
+  return message;
+}
```

Comment:
It's much simpler and safer to use std::error_code and related facilities to convert windows-specific error codes to std::error_code, then use .message() (returns a string) instead of FormatMessage. The narrowing conversion from DWORD to int is the only unsafe step, but all windows error codes fit in both, so it's fine. The codes go from 0 to 16k. https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes

```suggestion
std::string OsUtils::getWindowsErrorAsString(DWORD error_code) {
  return std::error_code{gsl::narrow<int>(error_code), std::system_category()}.message();
}
```